### ISSUE TITLE
Issue when using @ in comments (regression)

### ISSUE THREAD

**Issue by jsteel44** [kanboard/kanboard#3235]
### Actual behaviour

Typing @ in the comment field does not pop up a list of usernames. It used to work.

### Steps to reproduce

I'm using this patch to restore functionality (which was what it was some versions ago):

```
--- a/app/Template/task/show.php
+++ b/app/Template/task/show.php
@@ -54,7 +54,7 @@
 
 <?php if (!empty($comments)): ?>
     <?= $this->hook->render('template:task:show:before-comments', array('task' => $task, 'project' => $project)) ?>
-    <?= $this->render('task_comments/show', array(
+    <?= $this->render('comments/show', array(
         'task' => $task,
         'comments' => $comments,
         'project' => $project,
```

### Configuration

- Kanboard version: 1.0.43 (but I think I hit this issue around 1.0.40)
- Database type and version: 5.5.52-MariaDB
- PHP version:  5.4.16
- OS: Any
- Browser: Any

**Comment by rautamiekka** [comment]
I'd be updating that ancient PHP.

**Comment by jsteel44** [comment]
It's the latest in CentOS 7. But surely that's not the issue here? It worked before 1.0.40.

**Comment by NinjaPewPew** [comment]
We use Kanboard v1.0.41 and PHP 5.6.15 and the function still works.

**Comment by izno** [comment]
I'm also experiencing this issue: (worked fine in earlier versions)

These are my info from the 3 different installations:

**master** – Not working
**1.0.43** – Not working
**1.0.37** – Working

## Configuration (localhost running MAMP)

- Application version: master
- PHP version: 7.0.10
- PHP SAPI: apache2handler
- OS version: Darwin 14.5.0
- Database driver: sqlite
- Database version: 3.13.0
- Browser: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/58.0.3029.110 Safari/537.36

## Configuration (remote server)

Application version: 1.0.43
PHP version: 7.0.17
PHP SAPI: litespeed
OS version: Linux 2.6.32-673.26.1.lve1.4.27.el6.x86_64
Database driver: sqlite
Database version: 3.14.2
Browser: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/58.0.3029.110 Safari/537.36

## Configuration (remote server)
### OBS! Here "@" is working just fine
Application version: 1.0.37
PHP version: 7.0.17
PHP SAPI: litespeed
OS version: Linux 2.6.32-673.26.1.lve1.4.27.el6.x86_64
Database driver: sqlite
Database version: 3.14.2
Browser: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/58.0.3029.110 Safari/537.36

**Comment by jsteel44** [comment]
@izno does my patch fix your problem too?

**Comment by izno** [comment]
@jsteel44 ...I'm sorry which patch is that? Could you please include a link...

**Comment by izno** [comment]
@jsteel44 ...I'm sorry, but I can't really see where your patch is implemented! If I put it in the 

`app/Template/task/show.php`

Then it breaks the comment and no comment is showing. But I can see that the original file is ref. to another file:

`app/Template/task_comments/show.php`

Could you please describe how exactly you are using your patch?

**Comment by izno** [comment]
...now I have also tested it with "PHP version: 5.6.25" locally with MAMP and it still not showing the pop-up.

## Configuration
Application version: master
PHP version: 5.6.25
PHP SAPI: apache2handler
OS version: Darwin 14.5.0
Database driver: sqlite
Database version: 3.8.10.2
Browser: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/58.0.3029.110 Safari/537.36

**Comment by izno** [comment]
I have just tested... the mention massage is coming trough. So using the "@" in comments are working fine as functionality. But still the pop-up is not showing the list of users in the project group.

**Comment by jsteel44** [comment]
Maybe that's something to do with my issue; comments moved to task_comments and I've done upgrades in place so I still have the directory "comments". I'll have to try a fresh install and migrate my config to it.

**Comment by jsteel44** [comment]
Tried a fresh install and still have this issue. To restore functionality I must patch app/Template/task/show.php (as mentioned in my first comment) and then restore the app/Template/comments directory from a previous version. Now everything works OK.

**Comment by jsteel44** [comment]
This is a nicer fix. Works for me on a fresh install without having to recover old files.

```
--- a/app/Template/show.php	2017-05-19 13:26:39.988073234 +0100
+++ b/app/Template/show.php	2017-05-19 13:36:04.407520611 +0100
@@ -25,7 +25,11 @@
 
         <?php if ($editable): ?>
             <?= $this->render('task_comments/create', array(
-                'values'   => array(),
+                'values'   => array(
+                    'user_id' => $this->user->getId(),
+                    'task_id' => $task['id'],
+                    'project_id' => $task['project_id'],
+                ),
                 'errors'   => array(),
                 'task'     => $task,
             )) ?>
```

**Comment by izno** [comment]
@jsteel44 thanks for sharing...

I'm little confused... There is no file called

`app/Template/show.php`

Which file are patching?

**Comment by jsteel44** [comment]
@izno sorry it should have been app/Template/task_comments/show.php. Updated my last post to correct the path.

**Comment by izno** [comment]
@jsteel44 ...thanks! Works like a charm.

**Comment by fguillot** [comment]
The fix is available in Kanboard v1.0.44. Thanks.
